internal/parser/interface_test.go

```diff
@@ -0,0 +1,214 @@
+						Returns:           []models.Type{},
+						IsPointerReceiver: false,
+						IsExported:        true,
+						Comments:          []string{},
+					},
+				},
+			},
+		},
+		"interface with one method with comments": {
+			goCode: `
+			package main
+
+			type A interface {
+				//Foo does something
+				Foo()
+			}
+			`,
+			interfaceName: "A",
+			expectedInterface: models.Element{
+				Type: models.Type{Name: "main.A", InternalName: "A"},
+				Methods: []models.Method{
+					{
+						Name:              "Foo",
+						Params:            []models.Type{},
+						Returns:           []models.Type{},
+						IsPointerReceiver: false,
+						IsExported:        true,
+						Comments:          []string{"Foo does something"},
+					},
+				},
+			},
+		},
+		"interface with one method with params": {
+			goCode: `
+			package main
+			
+			type A interface {
+				Foo(a int, b string)
+			}
+			`,
+			interfaceName: "A",
+			expectedInterface: models.Element{
+				Type: models.Type{Name: "main.A", InternalName: "A"},
+				Methods: []models.Method{
+					{
+						Name:              "Foo",
+						Params:            []models.Type{{Name: "int", InternalName: "int"}, {Name: "string", InternalName: "string"}},
+						Returns:           []models.Type{},
+						IsPointerReceiver: false,
+						IsExported:        true,
+						Comments:          []string{},
+					},
+				},
+			},
+		},
+		"interface with one method with returns": {
+			goCode: `
+			package main
+
+			type A interface {
+				Foo() (int, string)
+			}
+			`,
+			interfaceName: "A",
+			expectedInterface: models.Element{
+				Type: models.Type{Name: "main.A", InternalName: "A"},
+				Methods: []models.Method{
+					{
+						Name:              "Foo",
+						Params:            []models.Type{},
+						Returns:           []models.Type{{Name: "int", InternalName: "int"}, {Name: "string", InternalName: "string"}},
+						IsPointerReceiver: false,
+						IsExported:        true,
+						Comments:          []string{},
+					},
+				},
+			},
+		},
+		"interface with a sub interface": {
+			goCode: `	
+			package main
+		
+			type A interface {
+				Foo() (int, string)
+			}
+			
+			type B interface {
+				// A is a sub interface
+				A
+				Bar()
+			}
+			`,
+			interfaceName: "B",
+			expectedInterface: models.Element{
+				Type: models.Type{Name: "main.B", InternalName: "B"},
+				Methods: []models.Method{
+					{
+						Name:              "Bar",
+						Params:            []models.Type{},
+						Returns:           []models.Type{},
+						IsPointerReceiver: false,
+						IsExported:        true,
+						Comments:          []string{},
+					},
+					{
+						Name:              "Foo",
+						Params:            []models.Type{},
+						Returns:           []models.Type{{Name: "int", InternalName: "int"}, {Name: "string", InternalName: "string"}},
+						IsPointerReceiver: false,
+						IsExported:        true,
+						Comments:          []string{" A is a sub interface"},
+					},
+				},
+			},
+		},
```

Comment:
Can you test with:
+ empty interface
+ imported named types
+ multi-line comments
+ non-exported methods
+ a sub interface with a name